# Testingcode.

### Depth measurements with realseanse D435
### T-shirts on hangers.

In [173]:
import pyrealsense2 as rs
import numpy as np
import cv2

In [174]:
output_dir = "output_and_screenshots"

### Testnumber

In [175]:
nr = 6

### Distance to first t-shirt

In [176]:
distance = "30cm"

### Alpha value

In [177]:
alpha = 0.3

### Take color and depth image.

In [178]:
pipe = rs.pipeline()
profile = pipe.start()

try:
    frames = pipe.wait_for_frames()
    for f in frames:
        print(f.profile)
        print(f.get_profile().format)

        # save depth img
        depth_frame = f.as_depth_frame()
        if depth_frame:
            depth_img = np.asanyarray(depth_frame.get_data())
            depth_img = cv2.applyColorMap(cv2.convertScaleAbs(depth_img, alpha = alpha), cv2.COLORMAP_JET)
            cv2.imwrite(f"{output_dir}/{nr}_depthimg_distance={distance}_alpha={alpha}.png", depth_img)

        # save color img
        color_frame = f.as_video_frame()
        if color_frame and f.profile.stream_type() == rs.stream.color:
            color_img = np.asanyarray(color_frame.get_data())
            color_img = cv2.cvtColor(color_img, cv2.COLOR_BGR2RGB)
            cv2.imwrite(f"{output_dir}/{nr}_colorimg_distance={distance}.png", color_img)
            break

finally:
    pipe.stop()

<pyrealsense2.[video_]stream_profile: Depth(0) 848x480 @ 30fps Z16>
<bound method PyCapsule.format of <pyrealsense2.[video_]stream_profile: Depth(0) 848x480 @ 30fps Z16>>
<pyrealsense2.[video_]stream_profile: Color(0) 640x480 @ 30fps RGB8>
<bound method PyCapsule.format of <pyrealsense2.[video_]stream_profile: Color(0) 640x480 @ 30fps RGB8>>


# Textfile with depthvalues

### Default shape

- color_img.shape = (480, 640, 3)
- depth_img.shape = (480, 848)

In [179]:
image_width = 320
image_height = 240

In [180]:
depth_width = 424
depth_height = 240

In [181]:
pipe = rs.pipeline()
profile = pipe.start()

try:
    frames = pipe.wait_for_frames()
    for f in frames:
        print(f.profile)
        print(f.get_profile().format)

        depth_frame = f.as_depth_frame()
        if depth_frame:
            depth_img = np.asanyarray(depth_frame.get_data())

            depth_img_resized = cv2.resize(depth_img, (depth_width, depth_height))

            depth_str = '\n'.join(['\t'.join([f"{str(val).zfill(4)}" if val != 0 else '    ' for val in row]) for row in depth_img_resized])
            
            print("Depth values:\n", depth_str)

            # save depth .txt
            with open(f"{output_dir}/{nr}_depthvalues_distance={distance}_dim={depth_width}x{depth_height}.txt", "w") as depth_file:
                depth_file.write(depth_str)

            # save depth .png
            depth_img_resized = cv2.applyColorMap(cv2.convertScaleAbs(depth_img_resized, alpha=alpha), cv2.COLORMAP_JET)
            cv2.imwrite(f"{output_dir}/{nr}_depthimg_distance={distance}_dim={depth_width}x{depth_height}_alpha={alpha}.png", depth_img_resized)
            break

finally:
    pipe.stop()

<pyrealsense2.[video_]stream_profile: Depth(0) 848x480 @ 30fps Z16>
<bound method PyCapsule.format of <pyrealsense2.[video_]stream_profile: Depth(0) 848x480 @ 30fps Z16>>
Depth values:
     	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	    	0260	0347	0347	0347	0347	0347	0347	0346	0346	0346	0346	0346	0346	0346	0346	0347	0347	0347	0346	0259	    	0085	0338	0337	0336	0335	0335